spyderlib/widgets/ipython.py

```diff
@@ -628,6 +632,23 @@ def clear_line(self):
     def clear_console(self):
         """Clear the whole console"""
         self.shellwidget.execute("%clear")
+        
+    @Slot()
+    def reset_namespace(self):
+        """Resets the namespace by removing all names defined by the user"""
+        
+        reply = QMessageBox.question(
+            None,
+            _("Reset IPython namespace"),
+            _("All user-defined variables will be removed."
+            "<br>Are you sure you want to clean the namespace?"),
+            QMessageBox.Yes | QMessageBox.No,
+            )
+
+        if reply == QMessageBox.No:
+            return
+        
+        self.shellwidget.execute("%reset -f")

```

Comment:
I don't like having return in the middle of nowhere (I know I used that as an example... I was being lazy sorry)

It would be better to have

```python
       if reply == QMessageBox.Yes:
           self.shellwidget.execute("%reset -f")       
```